parsexml.py

```diff
@@ -9,39 +9,82 @@
-                    return
+            records = doc['main']['DATA_RECORD']
+            for x in records:
+                docid = x.get('DOCUMENT_IMAGE_ID',x.get('DOCUMENT_NM'))
+                if type == ".xml":
+                    txtfn = os.path.join(os.path.dirname(fn),'PDF_image',docid+'.txt')
+                    if os.path.isfile(txtfn):
+                        x['LAST_MODIFIED_TS'] = format_date('LAST_MODIFIED_TS',x)
+                        x['PATENT_ISSUE_DT'] = format_date('PATENT_ISSUE_DT',x)
+                        x['DECISION_MAILED_DT'] = format_date('DECISION_MAILED_DT',x)
+                        x['PRE_GRANT_PUBLICATION_DT'] = format_date('PRE_GRANT_PUBLICATION_DT',x)
+                        x['APPLICANT_PUB_AUTHORIZATION_DT'] = format_date('APPLICANT_PUB_AUTHORIZATION_DT',x)
+                        x['appid'] = x.pop('BD_PATENT_APPLICATION_NO')
+                        x['doc_date'] = x.pop('DOCUMENT_CREATE_DT')
+                        with open(txtfn) as dr:
+                            text = dr.read()
+                            x['textdata'] = text
+                    else:
+                        logging.info("TXT file: "+docid+".txt  does not exist. JSON file creation skipped.")
+                        return
+                if (args.solr):
+                    jsontext = json.dumps(x)
+                    #Send content to Solr to update index
+                    jsontext = '{"add":{ "doc":'+jsontext+',"boost":1.0,"overwrite":true, "commitWithin": 1000}}'
+                    with open(os.path.join(os.path.dirname(fname),'jsoncomplete.txt'),'a+') as logfile:
+                        logfile.seek(0)
+                        if docid+"\n" in logfile:
+                            logging.info("-- file: "+docid+"  already processed by Solr")
+                            continue
+                        else:
+                            logging.info("-- Sending file: "+docid+" to Solr")
+                            url = "http://54.208.116.77:8983/solr/ptab/update"
+                            headers = {"Content-type" : "application/json"}
+                            r = requests.post(url, data=jsontext, headers=headers)
+                            rdict = json.loads(r.text)
+                            status = rdict["responseHeader"]["status"] 
+                            if status == 0:
+                                logging.info("-- Solr update for file: "+docid+" complete")
+                            else:
+                                logging.info("-- Solr error for doc: "+docid+" error: "+', '.join("{!s}={!r}".format(k,v) for (k,v) in rdict.items()))
+                            
             #transform output to json and save to file with same name
-            with open(fn,'w') as outfile:
-                json.dump(doc,outfile)
-                logging.info("-- Processing of XML file and creation of json file complete")
+            if type == ".xml":
```

Comment:
It's a little hard to follow why it's writing a JSON file way at the end here.

1. These sections should be broken out into functions, so the overall steps are easier to read.
2. The basic conversion and save should be done before the load into Solr, so if anything goes wrong during the load step, then at least you already have a saved file to start back up from.